### ISSUE TITLE
Getting intermittent connection timeouts in us-west-2 with SQS

### ISSUE THREAD

**Issue by efenderbosch** [aws/aws-sdk-java#1195]
This showed up in our QA environment a few days ago. We're using SDK 1.11.147 and update it regularly.

```com.amazonaws.SdkClientException: Unable to execute HTTP request: Connect to sqs.us-west-2.amazonaws.com:443 [sqs.us-west-2.amazonaws.com/54.240.249.101] failed: connect timed out```

Any ideas what I can do to debug this? At first I thought this was a permission problem, so I added a queue policy to our cloud formation. Here's the snippet from our serverless.yml:

```yaml
    someQueuePolicy:
      Type: AWS::SQS::QueuePolicy
      Properties:
        PolicyDocument:
          Id: "${self:custom.stage}-${self:service}"
          Version: '2012-10-17'
          Statement:
          - Sid: "${self:custom.stage}-${self:service}"
            Action:
            - sqs:ChangeMessageVisibility
            - sqs:ChangeMessageVisibilityBatch
            - sqs:DeleteMessage
            - sqs:DeleteMessageBatch
            - sqs:GetQueueAttributes
            - sqs:GetQueueUrl
            - sqs:ListDeadLetterSourceQueues
            - sqs:ListQueues
            - sqs:ReceiveMessage
            - sqs:SendMessage
            - sqs:SendMessageBatch
            Effect: Allow
            Principal:
              AWS:
              - arn:aws:iam::<our lambda role>
              - arn:aws:iam::<our ecs role>
        Queues:
          - Ref: someQueue
```

Anything else I should be looking at? This lambda is deployed in non-default security groups and subnets. At least one of the security groups has outbound 443 enabled.

**Comment by dagnir** [comment]
Hi @efenderbosch this seems like a network issue and is unrelated to the SDK. Do you see this problem regularly? If so, you may want to open through the console to see if it's a service side issue with SQS.

**Comment by efenderbosch** [comment]
Sorry about the delayed reply. I opened a support issue as well and got some information there. We're doing some initialization outside of (before) the handler method (injection via Guice, etc.) and that may not work when connecting to network resources when deploying in the non-default lambda VPC. The ENI isn't guaranteed to be available until the handler method has been invoked. Basically we had to delay some network lookups (like SQS getQueueURL) until the handler has been invoked, instead of performing that during initialization.
